### PR TITLE
pkg/manifests: Configure remote write more conservatively

### PR DESCRIPTION
**Pull request by brancz**
Note this PR does not re-enable telemetry by default, it merely if enabled fine tunes the remote write config. With this config we neither require resharding nor does remote write fall behind trying to replicate the data to the remote which did happen with the previous configuration. Additionally this drives the req/s per Prometheus down to ~4req/s.

@LiliC @s-urbaniak @paulfantom @simonpasquier @pgier

### CODE REVIEW

File: pkg/manifests/manifests.go [hunk 1]
```diff
@@ -1129,15 +1129,24 @@ func (f *Factory) PrometheusK8s(host string, grpcTLS *v1.Secret, trustedCABundle
...
-				MaxShards: 100,
-				// Default: 30ms
-				MinBackoff: "500ms",
-				// Default: 100ms
-				MaxBackoff: "2s",
+				// Amount of samples to load from the WAL into the in-memory
+				// buffer before waiting for samples to be sent successfully
+				// and then continuing to read from the WAL.
+				Capacity: 30000,
+				// Should we accumulate 10000 samples before the batch send
```
Comment (LiliC) [inline pkg/manifests/manifests.go]: You mean less than 1000 samples?

Comment (brancz) [reply pkg/manifests/manifests.go]: The comment refers to the `MaxSamplesPerSend`, which is set to 10000.

**Comment by paulfantom** [comment]
The generation step is failing. Putting this on hold

/hold

File: pkg/manifests/manifests.go [hunk 2]
```diff
@@ -1129,15 +1129,24 @@ func (f *Factory) PrometheusK8s(host string, grpcTLS *v1.Secret, trustedCABundle
...
+				MaxSamplesPerSend: 10000,
+				// Batch samples for 1m until we send them if we not reach the
+				// 10000 MaxSamplesPerSend first.
+				BatchSendDeadline: "1m",
+				// Backoff is doubled on every backoff. We start with 1s
+				// backoff and double until the MaxBackOff.
+				MinBackoff: "1s",
+				// 128s is the 8th backoff in a row, once we end up here, we
+				// don't increase backoff time anymore. As we would at most
+				// produce (concurrency/256) number of requests per second.
```
Comment (cben) [inline pkg/manifests/manifests.go]: These comments are very helpful for uninitated readers :+1:
Couple questions that may be worth clarifying:
- 256s is more that 1m.  May multiple backing-off batches overlap, or do they block later batches until success / give up?
- "we don't increase backoff time anymore" — does that mean retries continues indefinitely at 256s intervals?  Or is the batch given up after 8 failures?

Comment (brancz) [reply pkg/manifests/manifests.go]: 1) they have to be consecutive so only the concurrency factor plays a role here.
2) it retries infinitely at 256s intervals and stops after the WAL is cut which happens every 2 hours. Then the mechanisms starts tailing the “new” WAL.

**Comment by LiliC** [comment]
@brancz you need bugzilla for this now, just create one with "Alerts for remote write firing" 🙄

**Comment by brancz** [comment]
It’s ok this is fine to wait until 4.5 master opens again.

**Pull request merged**